### DISCUSSION TITLE
hand robot configuration

### DISCUSSION THREAD

**Discussion by Spphire** [NVlabs/curobo#52] [Software Q&A] [answered]
To configure a hand robot, should I create configs for each finger?

**Comment by balakumar-s** [comment]
You can use a single file for the full robot hand.

Here is an example: https://github.com/NVlabs/curobo/blob/main/src/curobo/content/configs/robot/iiwa_allegro.yml

**Comment by Spphire** [comment]
https://github.com/NVlabs/curobo/blob/main/src/curobo/content/configs/robot/iiwa_allegro.yml#L35

I noticed that ee_link only set one target, does that mean it can be only controlled one finger?

**Reply by balakumar-s** [reply, marked as answer]
You can add any additional targets to link_names in the configuration file. Then you can pass in targets for the additional link names like here: https://github.com/NVlabs/curobo/blob/4aabfa020601f9c50e34df6517e0916faf0dfd32/examples/isaac_sim/multi_arm_reacher.py#L301

You can try out the `iiwa_allegro.yml` file in the multi-arm example by running `omni_python examples/isaac_sim/multi_arm_reacher.py --robot iiwa_allegro.yml`

**Comment by Spphire** [comment]
https://github.com/NVlabs/curobo/assets/56157591/5a7e8062-e1b1-490d-91b7-2475d0ddbfbf

Is it normal that "Plan did not converge to a solution" appeared frequently?

**Reply by balakumar-s** [reply]
We haven't tested on tracking more than two end-effectors.  The weights for this cost might need to tuned (lowered) for tracking 5 targets: https://github.com/NVlabs/curobo/blob/d85ae41fba22672d8178e3bed2af464119936e1b/src/curobo/content/configs/task/gradient_trajopt.yml#L35

**Reply by Spphire** [reply]
what are the means of these 4 values?

**Reply by balakumar-s** [reply]
They refer to the relative weighting between orientation and position.

We re-tuned the weights and it should work better for tracking multiple end-effectors. Try our latest release and let us know if you still have issues in tracking multiple end-effectors.
